haystack/components/generators/openai.py

```diff
@@ -170,6 +170,7 @@ def from_dict(cls, data: Dict[str, Any]) -> "OpenAIGenerator":
     def run(
         self,
         prompt: str,
+        run_time_system_prompt: Optional[str] = None,
```

Comment:
hmm is it possible for us to leave the name as `system_prompt` and still keep the behavior you mention in the docs? 

I think it lines up better with our design of other components whose runtime variables optionally override the init ones. E.g. we do this regularly in our retrievers with the `top_k` parameter, but we still use the same name `top_k` both in the init and run methods.